fast_matched_filter/src/matched_filter.c

```diff
@@ -102,15 +101,37 @@ float network_corr(float *templates, float *sum_square_template, int *moveouts,
  
 //-------------------------------------------------------------------------
 float corrc(float *templates, float sum_square_template,
-            float *data, double *csum_square_data,
+            float *data, float *csum_square_data,
             int n_samples_template) {
 
     int i;
-    float numerator = 0, sum_square_data;
+    float numerator = 0, denominator = 0, cc = 0;
     
-    for (i = 0; i < n_samples_template; i++) numerator += templates[i] * data[i];
-    sum_square_data = (float)(csum_square_data[i] - csum_square_data[0]); // note i == n_samples_template
-   
-    return numerator / sqrtf(sum_square_template * sum_square_data);
+    for (i = 0; i < n_samples_template; i++){
+        numerator += templates[i] * data[i];
+    }
+    denominator = sum_square_template * csum_square_data[0];
+    //denominator = sum_square_template * denominator;
+    if (denominator > 0.00001) cc = numerator / sqrt(denominator);
```

Comment:
I think this will result in datasets with small variances not being normalised.  I have been running into this issue [here](https://github.com/eqcorrscan/EQcorrscan/pull/182), but we use a double for the normalisation then cast back to float.  It doesn't seem to add much memory overhead (you are only storing one double in memory, and casting to float), but it may increase the time.  It might be worth a test because it will allow you to apply these codes to a greater range of data.
I also think it is worth making the user aware that their variance is too small and that their correlations are not being normalised when the denominator is < 0.00001.
This is a rare case for passive seismic data, so up to you, but worth making sure that any users know the limitations.